src/sentry/grouping/fingerprinting/__init__.py

```diff
@@ -410,27 +313,35 @@ def text(self):
 
 
 class Rule:
-    def __init__(self, matchers, fingerprint, attributes, is_builtin: bool = False):
+    def __init__(
+        self,
+        matchers: Sequence[Match],
+        fingerprint: str,
+        attributes: Sequence[Any],
```

Comment:
I can do some inspection tomorrow if you're not satisfied with the use of `Any`.